src/components/button-swap/button-swap.test.ts

```diff
@@ -0,0 +1,185 @@
+import path from 'path';
+import type { ElementHandle, PleasantestUtils } from 'pleasantest';
+import { withBrowser } from 'pleasantest';
+import { loadTwigTemplate, loadGlobalCSS } from '../../../test-utils';
+
+// Helper to load the Twig template file
+const componentMarkup = loadTwigTemplate(
+  path.join(__dirname, './button-swap.twig')
+);
+
+// Helper to initialize the component JS
+const initJS = (
+  utils: PleasantestUtils,
+  buttonSwapEl: ElementHandle,
+  firstBtnCallback?: () => void,
+  secondBtnCallback?: () => void
+) =>
+  utils.runJS(
+    `
+    import { initButtonSwap } from './button-swap'
+    export default (buttonSwapEl, firstBtnCallback, secondBtnCallback) => {
+      initButtonSwap(buttonSwapEl, {
+        firstBtnCallback,
+        secondBtnCallback
+      })
+    }
+    `,
+    [buttonSwapEl, firstBtnCallback, secondBtnCallback]
+  );
+
+test(
+  'Initial state',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(await componentMarkup());
+    await loadGlobalCSS(utils);
+
+    // Visually hidden text for a more inclusive UX
+    const statusMsg = await screen.getByRole('status');
+    await expect(statusMsg).toHaveTextContent(
+      /^unsubscribed from notifications$/i
+    );
+
+    const statusMsgs = await screen.getAllByRole('status');
+    expect(statusMsgs.length).toBe(1);
+
+    const firstBtn = await screen.getByRole('button', {
+      name: /^get notifications$/i,
+    });
+    await expect(firstBtn).toBeVisible();
+
+    const secondBtn = await screen.queryByRole('button', {
+      name: /^turn off notifications$/i,
+    });
+    expect(secondBtn).toBeNull();
+  })
+);
+
+test(
+  'Swap UI state when clicked',
+  withBrowser(async ({ utils, screen, user }) => {
+    await utils.injectHTML(await componentMarkup());
+    await loadGlobalCSS(utils);
+    // I'd like to avoid using a test ID, but I couldn't figure out a different way.
+    // @todo Can this be done without at test ID?
+    await initJS(utils, await screen.getByTestId('test-id'));
```

Comment:
@cloudfour/dev Any thoughts? Suggestions?

Reply:
Could we use `getByRole('region')` or something to grab the `section`?

Reply:
If you end up adding an accessible name to the section, the name can be used to filter as well:

```js
await screen.getByRole('region', { name: /something/i })
```

Reply:
> Could we use `getByRole('region')` or something to grab the `section`?

@calebeby @Paul-Hebert I tried the following initially:
```js
await initJS(utils, await screen.getByRole('region'));
```
But that would result in this failed test:
```
Unable to find an accessible element with the role "region"

    Here are the accessible roles:

      document:

      Name "":
      <body>
      <section data-testid="test-id" class="c-button-swap js-c-button-swap">[...]</section>


    </body>

      --------------------------------------------------
      status:

      Name "":
      <span
      tabindex="-1"
      role="status"
      class="u-hidden-visually js-c-button-swap__message"
    >

          Unsubscribed from notifications

    </span>

      --------------------------------------------------
      button:

      Name "Get notifications":
      <button class="c-button js-c-button-swap__button">[...]</button>

      --------------------------------------------------

    Within: #document
```

Reply:
Oh, maybe I need to give it a name:
> Using the `<section>` element will automatically communicate a section has a role of `region` if it is given an accessible name. Developers should always prefer using the correct semantic HTML element, in this case `<section>`, over using ARIA.

Source: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Region_role#description